plugins/Installation/HostPortExtractor.php

```diff
@@ -0,0 +1,182 @@
+     *
+     * @param string $dbHost The user provided database server host
+     * @return boolean Whether the provided host is correct IPv6
+     */
+    private static function isIPv6(string $dbHost): bool
+    {
+        // IPv6 addresses must contain '[' & ']' to be considered valid
+        if (strpos($dbHost, '[') !== false && strpos($dbHost, ']') !== false) {
+            // filter_var requires potential IPv6 addresses to not be encased
+            preg_match_all('/\[(.*?)\]/', $dbHost, $matches);
+            $listOfTextInsideSquareBrackets = $matches[1];
+
+            /*
+             * Only return true if there is some text inside square brackets,
+             * and that text is considered valid IPv6 as per filter_var()
+             */
+            if (
+                count($listOfTextInsideSquareBrackets) > 0 &&
+                filter_var(
+                    $listOfTextInsideSquareBrackets[0],
+                    FILTER_VALIDATE_IP,
+                    FILTER_FLAG_IPV6
+                ) !== false
+            ) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Extracts the Host and Port from a user provided database server host,
+     * assuming the provided host is a IPv6 address, with or without a port
+     *
+     * @param string $dbHost The user provided database server host
+     * @throws InvalidArgumentException if the provided dbHost is not a valid IPv6
+     * @return HostPortExtractor The extracted Host & Port
+     */
+    private static function extractIPv6(string $dbHost): HostPortExtractor
+    {
+        if (!self::isIPv6($dbHost)) {
+            throw new InvalidArgumentException('$dbHost must be a valid IPv6 address');
+        }
+
+        // find and extract the text inside square brackets: []
+        preg_match_all('/\[(.*?)\]/', $dbHost, $matches);
+        $listOfTextInsideSquareBrackets = $matches[1];
+
+        $port = '';
+
+        // extract text after closing bracket to search for port
+        $components = explode(']', $dbHost);
+        if (count($components) > 1) {
+            // check if the text is a valid port e.g. ':3000'
+            preg_match('/^(:\d+)$/', $components[1], $portMatches);
+            if (count($portMatches) > 0) {
+                // only save the port number, not the colon
+                $port = substr($portMatches[1], 1);
+            }
+        }
+        // db connector requires IPv6 to be encased in square brackets
+        $host = '[' . $listOfTextInsideSquareBrackets[0] . ']';
+
+        return new HostPortExtractor($host, $port);
+    }
+
+    /**
+     * Determines if the provided host is a Unix Socket
+     *
+     * @param string $dbHost The user provided database server host
+     * @return boolean Whether the provided host is a unix socket
+     */
+    private static function isUnixSocket(string $dbHost): bool
+    {
+        return (strpos($dbHost, '/') !== false);
```

Comment:
Is the check meant to be that the string starts with a slash? Or that it has a slash anywhere?
```suggestion
        return (strpos($dbHost, '/') === 0);
```

Reply:
This is the original logic so I've changed it as little as possible, however yes the string should be required to start with a slash, to match "/unix/socket" format.

Reply:
I see re: the move of the code without changing the logic.
I think that's a valid small improvement we can do here unless sockets can be defined in some other way where there's no leading `/` and it's prefixed with something else before that. Let's keep it as is.